sparkplug-core/src/java/sparkplug/function/SerializableFn.java

```diff
@@ -28,33 +45,99 @@ private SerializableFn() {
     }
 
 
+    /**
+     * Construct a new serializable wrapper for the function with an explicit
+     * set of required namespaces.
+     *
+     * @param fn Clojure function to wrap
+     * @param namespaces collection of namespaces required
+     */
+    protected SerializableFn(IFn fn, Collection<String> namespaces) {
+        this.f = fn;
+        List<String> namespaceColl = new ArrayList<String>(namespaces);
+        Collections.sort(namespaceColl);
+        this.namespaces = Collections.unmodifiableList(namespaceColl);
+    }
+
+
+    /**
+     * Serialize the function to the provided output stream.
+     * An unspoken part of the `Serializable` interface.
+     *
+     * @param out stream to write the function to
+     */
     private void writeObject(ObjectOutputStream out) throws IOException {
-        // Attempt to derive the needed Clojure namespace from the function's class name.
-        String namespace = Compiler.demunge(this.f.getClass().getName()).split("/")[0];
-        out.writeObject(namespace);
-        out.writeObject(this.f);
+        try {
+            logger.trace("Serializing " + f);
+            // Write the function class name.
```

Comment:
Looks like it's only writing the class name for debugging/logging purposes, worth noting that in a comment.